src/DataBase/EntryDatabase.cs

```diff
@@ -139,69 +140,86 @@ public static bool Delete(Func<IEntryModel, bool> filter, EntryType entryType)
             return true;
         }
 
-        public static bool DeleteList(Func<IEntryModel, bool> filter, EntryType entryType)
+        public async Task<bool> DeleteListAsync(Expression<Func<IEntryModel, bool>> filter, EntryType entryType)
         {
-            var db = new DatabaseContext();
-
             if (entryType == EntryType.Income)
             {
-                var entriesToRemove = db.Incomes.Where(filter).Cast<IncomeModel>().ToList();
-                db.Incomes.RemoveRange(entriesToRemove);
+                var entriesToRemove = await Database.Incomes.Where(filter).Cast<IncomeModel>().ToListAsync();
+                Database.Incomes.RemoveRange(entriesToRemove);
             }
             else
             {
-                var entriesToRemove = db.Expenses.Where(filter).Cast<ExpenseModel>().ToList();
-                db.Expenses.RemoveRange(entriesToRemove);
+                var entriesToRemove = await Database.Expenses.Where(filter).Cast<ExpenseModel>().ToListAsync();
+                Database.Expenses.RemoveRange(entriesToRemove);
             }
 
-            db.SaveChanges();
+            await Database.SaveChangesAsync();
             return true;
         }
 
-        public static bool DeleteList(IEnumerable<int> idArray, int userId, EntryType entryType)
+        public async Task<bool> DeleteListAsync(IEnumerable<int> idArray, int userId, EntryType entryType)
         {
-            var db = new DatabaseContext();
-
             if (entryType == EntryType.Income)
             {
-                var entriesToRemove = idArray.Select(id => db.Incomes.FirstOrDefault(x => x.Id == id && x.UserId == userId)).ToList();
-                db.Incomes.RemoveRange(entriesToRemove);
+                var list = new List<IncomeModel>();
+                foreach (var id in idArray)
+                {
+                    var temp = await Database.Incomes.FirstOrDefaultAsync(x => x.Id == id && x.UserId == userId);
+                    list.Add(temp);
+                }
+                Database.Incomes.RemoveRange(list);
```

Comment:
Why was this changed from LINQ to foreach?

Reply:
Wanted to write this as quick possible, and wasn't sure how to change this expression to be async. Spent some time and figured out how to write as a linq expression, so this was a good question to ask :D